dbt/include/bigquery/macros/materializations/table.sql

```diff
@@ -110,14 +110,16 @@ df.write \
   .mode("overwrite") \
   .format("bigquery") \
   .option("writeMethod", "indirect").option("writeDisposition", 'WRITE_TRUNCATE') \
+  {%- if partition_config is not none %}
   {%- if partition_config.data_type | lower in ('date','timestamp','datetime') %}
   .option("partitionField", "{{- partition_config.field -}}") \
   {%- if partition_config.granularity is not none %}
-  .option("partitionType", "{{- partition_config.granularity -}}") \
+  .option("partitionType", "{{- partition_config.granularity|upper -}}") \
```

Comment:
If we're space-padding the pipe on line 122 (and like it is on line 114), then I would do it here too.